fbench/src/fbench/fbench.cpp

```diff
@@ -250,24 +243,24 @@ FBench::PrintSummary()
     printf("minimum response time:  %8.2f ms\n", status._minTime);
     printf("maximum response time:  %8.2f ms\n", status._maxTime);
     printf("average response time:  %8.2f ms\n", status.GetAverage());
-    if (p25 > status._timetable.size() / status._timetableResolution - 1)
-        printf("25 percentile:          %8.2f ms (approx)\n", p25);
-    else         printf("25 percentile:          %8.2f ms\n", p25);
-    if (p50 > status._timetable.size() / status._timetableResolution - 1)
-        printf("50 percentile:          %8.2f ms (approx)\n", p50);
-    else         printf("50 percentile:          %8.2f ms\n", p50);
-    if (p75 > status._timetable.size() / status._timetableResolution - 1)
-        printf("75 percentile:          %8.2f ms (approx)\n", p75);
-    else         printf("75 percentile:          %8.2f ms\n", p75);
-    if (p90 > status._timetable.size() / status._timetableResolution - 1)
-        printf("90 percentile:          %8.2f ms (approx)\n", p90);
-    else         printf("90 percentile:          %8.2f ms\n", p90);
-    if (p95 > status._timetable.size() / status._timetableResolution - 1)
-        printf("95 percentile:          %8.2f ms (approx)\n", p95);
-    else         printf("95 percentile:          %8.2f ms\n", p95);
-    if (p99 > status._timetable.size() / status._timetableResolution - 1)
-        printf("99 percentile:          %8.2f ms (approx)\n", p99);
-    else         printf("99 percentile:          %8.2f ms\n", p99);
+
+    for (double percentile : {25.0, 50.0, 75.0, 90.0, 98.0, 99.0, 99.5, 99.6, 99.7, 99.8, 99.9}) {
+        if (percentile <= 99.0) {
+            //Backwards compatible printing
+            if (percentile > (status._timetable.size() / status._timetableResolution - 1)) {
```

Comment:
you need to check the result of the percentile calculation against the sampled data to see if it is approximate, not the input parameter of the percentile calculation...

Reply:
Good catch, fixed